### PR TITLE
nix-janitor: 0.2.0 -> 0.3.1

### PR DESCRIPTION
**Pull request by NobbZ**
## Description of changes

* https://github.com/NobbZ/nix-janitor/releases/tag/0.3.0
* https://github.com/NobbZ/nix-janitor/releases/tag/0.3.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ni/nix-janitor/package.nix [hunk 1]
```diff
@@ -2,21 +2,32 @@
...
-    hash = "sha256-nGtbBNU6xFWXnmL1AaUbSpO0z5Kq2t/Mn8sqwzjNlkE=";
+    hash = "sha256-xoVByI17rt2SCY3ULg12S8QsoXGhQWZlOpPpK2mfcPY=";
   };
 
-  cargoHash = "sha256-j3i4c3KjI8ehg42FqbPp+8M15zT9Bu76P4zv8ApUoeA=";
+  cargoHash = "sha256-QG2hHM4KBSU6+droew2WnOFxWRTpk9griIPMD8MLSbw=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
```
Comment (OPNA2608) [inline pkgs/by-name/ni/nix-janitor/package.nix:24]:
```suggestion
  postInstall = lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) ''
```

File: pkgs/by-name/ni/nix-janitor/package.nix [hunk 2]
```diff
@@ -2,21 +2,32 @@
...
+    for shell in bash fish zsh; do
+      completionFile=janitor.$shell
+      $out/bin/janitor --completions $shell > $completionFile
```
Comment (OPNA2608) [inline pkgs/by-name/ni/nix-janitor/package.nix:27]:
```suggestion
      ${stdenv.hostPlatform.emulator buildPackages} $out/bin/janitor --completions $shell > $completionFile
```

File: pkgs/by-name/ni/nix-janitor/package.nix [hunk 3]
```diff
@@ -2,21 +2,32 @@
   lib,
```
Comment (OPNA2608) [inline pkgs/by-name/ni/nix-janitor/package.nix:2]:
```suggestion
  stdenv,
  buildPackages,
  lib,
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
To not break cross, you need to check if you can execute `$out/bin/janitor` during the build - either natively or via an emulator.

**Comment by NobbZ** [comment]
Thanks for showing this to me @OPNA2608, I was not aware of this being possible.

Would you mind if we do not do that change in this PR, but instead I will revise all packages maintained by me and check which else have to be modified this way?

**Comment by OPNA2608** [comment]
That's okay I guess... Feel free to request a review from me on the follow-up PR.

**Review by OPNA2608 — APPROVED** [review]

**Pull request merged**